### DISCUSSION TITLE
Section 3.13 Variáveis

### DISCUSSION THREAD

**Discussion by eduardocortez3** [chgasparoto/curso-aws-com-terraform#95] [Q&A]
Boa tarde Gasparoto.

Uma dúvida.
Qual a melhor estratégia para criar as variáveis de ambiente de uma aplicação? Por exemplo, URL's, Token's, Portas, e etc.
Tenho uma necessidade com o Elastic Benastalk da AWS, meu arquivos TF estão rodando direitinho criando o ambiente.
A nossa aplicação utiliza várias variáveis, eu gostaria de colocar o terraform para criar essas variáveis e gerenciá-las.
Estou utilizando o resource aws_elastic_beanstalk_environment e nele temos o campo setting {} aonde configuramos o ambiente do EB, entre outras coisas, as variáveis de ambiente da aplicação, segue abaixo um exemplo:

```hcl
setting {
    namespace = "aws:elasticbeanstalk:application:environment"
    name      = "VAR-A"
    value     = "https://XXX.xxx.com.br/api/v1"
    resource  = ""
  }
setting {
    namespace = "aws:elasticbeanstalk:application:environment"
    name      = "VAR-B"
    value     = "X-Y-Z"
    resource  = ""
  }
setting {
    namespace = "aws:elasticbeanstalk:application:environment"
    name      = "VAR-C"
    value     = "W-W-W"
    resource  = ""
  }
```

E assim por diante.... Veja que é repetitivo. Minha dúvida é, como colocar isso no terraform? Não queria lançar isso no mesmo arquivos tf do beanstalk e o arquivo variables.tf está ficando muito grande.

Att.

**Comment by chgasparoto** [comment]
@eduardocortez3 você pode ter uma única variável do tipo `object` e iterar sobre esse objeto utilizando o `dynamic` block.

Aqui tem um exemplo de como utilizá-lo: https://www.terraform.io/language/expressions/dynamic-blocks

**Reply by eduardocortez3** [reply]
Obrigado, vou olhar essa sugestão.
